src/extra/components/xgm.py

```diff
@@ -353,7 +361,36 @@ def pulse_energy(self, sase=None):
 
             self._pulse_energy[pg] = pulse_energy
 
-        return self._pulse_energy[pg]
+        pulse_energy = self._pulse_energy[pg]
+        if series:
+            # Use .where() to convert 0's to nans, then convert to a Series,
+            # then drop the nan entries.
+            pulse_energy = pulse_energy.where(pulse_energy > 0).to_series().dropna()
```

Comment:
units are lost at that point, you need to add them back.

Reply:
Fixed in 3ad2174.

Reply:
By chance I came across a run with negative XGM values. I dislike magic values for filtering, but at least we shoudl go for `== 0`.

Reply:
Fixed in 8bed29a.